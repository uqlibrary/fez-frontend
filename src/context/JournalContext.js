import React from 'react';

export default React.createContext({
    journalDetails: null,
    jnlDisplayType: null,
});
