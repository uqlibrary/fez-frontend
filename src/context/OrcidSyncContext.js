import { createContext } from 'react';

export const OrcidSyncContext = createContext({
    orcidSyncProps: {},
});

export default OrcidSyncContext;
