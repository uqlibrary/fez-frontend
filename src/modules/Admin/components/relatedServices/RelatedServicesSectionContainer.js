import React from 'react';
import PropTypes from 'prop-types';
import Immutable from 'immutable';
import { connect } from 'react-redux';
import { getFormValues } from 'redux-form/immutable';

import { FORM_NAME } from '../../constants';
import { FormValuesContext } from 'context';
import RelatedServicesSection from './RelatedServicesSection';

export const RelatedServicesSectionContainer = ({ disabled, formValues }) => {
    return (
        <FormValuesContext.Provider value={{ formValues: formValues.toJS() }}>
            <RelatedServicesSection disabled={disabled} />
        </FormValuesContext.Provider>
    );
};

RelatedServicesSectionContainer.propTypes = {
    disabled: PropTypes.bool,
    formValues: PropTypes.object,
};

export const mapStateToProps = (state, ownProps) => {
    const formValues = getFormValues(FORM_NAME)(state) || Immutable.Map({});

    return {
        disabled: ownProps.disabled,
        formValues: formValues.get('relatedServicesSection') || Immutable.Map({}),
    };
};

export default connect(mapStateToProps)(React.memo(RelatedServicesSectionContainer));
