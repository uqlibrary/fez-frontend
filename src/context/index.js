import { useContext } from 'react';

import AccountContext from './AccountContext';
import EditableContext from './EditableContext';
import FormErrorsContext from './FormErrorsContext';
import FormValuesContext from './FormValuesContext';
import LocallyStoredReducerContext from './LocallyStoredReducerContext';
import OrcidSyncContext from './OrcidSyncContext';
import RecordContext from './RecordContext';
import RecordsSelectorContext from './RecordsSelectorContext';
import TabbedContext from './TabbedContext';
import ScrollToSectionContext from './ScrollToSectionContext';
import JournalContext from './JournalContext';

export const useAccountContext = () => useContext(AccountContext);
export const useEditableContext = () => useContext(EditableContext);
export const useFormErrorsContext = () => useContext(FormErrorsContext);
export const useFormValuesContext = () => useContext(FormValuesContext);
export const useLocallyStoredReducerContext = () => useContext(LocallyStoredReducerContext);
export const useOrcidSyncContext = () => useContext(OrcidSyncContext);
export const useRecordContext = () => useContext(RecordContext);
export const useRecordsSelectorContext = () => useContext(RecordsSelectorContext);
export const useTabbedContext = () => useContext(TabbedContext);
export const useScrollToSectionContext = () => useContext(ScrollToSectionContext);
export const useJournalContext = () => useContext(JournalContext);

export {
    AccountContext,
    EditableContext,
    FormErrorsContext,
    FormValuesContext,
    LocallyStoredReducerContext,
    OrcidSyncContext,
    RecordContext,
    RecordsSelectorContext,
    ScrollToSectionContext,
    TabbedContext,
    JournalContext,
};
