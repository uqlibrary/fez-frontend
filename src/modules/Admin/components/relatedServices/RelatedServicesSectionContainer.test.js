import { RelatedServicesSectionContainer, mapStateToProps } from './RelatedServicesSectionContainer';
import Immutable from 'immutable';

jest.mock('../../../../context');
import { useFormValuesContext } from 'context';

const setup = (testProps = {}, args = { isShallow: true }) => {
    const props = {
        formValues: {
            toJS: jest.fn(() => ({})),
        },
        ...testProps,
    };

    return renderComponent(RelatedServicesSectionContainer, props, args);
};

describe('RelatedServicesSectionContainer', () => {
    beforeEach(() => {
        useFormValuesContext.mockImplementation(() => ({
            formValues: {
                relatedServices: 'Test',
            },
        }));
    });

    afterEach(() => {
        useFormValuesContext.mockReset();
    });

    it('should render properly', () => {
        const render = setup({});

        expect(render.getRenderOutput()).toMatchSnapshot();
    });

    it('should map state to props', () => {
        expect(mapStateToProps({}, {})).toEqual({
            formValues: Immutable.Map({}),
        });
    });
});
