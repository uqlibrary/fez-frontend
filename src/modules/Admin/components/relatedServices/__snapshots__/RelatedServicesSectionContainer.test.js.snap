// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedServicesSectionContainer should render properly 1`] = `
<Context.Provider
  value={
    {
      "formValues": {},
    }
  }
>
  <Memo(RelatedServicesSection) />
</Context.Provider>
`;
